homeassistant/components/matter/entity.py

```diff
@@ -31,6 +37,23 @@
 LOGGER = logging.getLogger(__name__)
 
 
+def catch_matter_error[_R, **P](
+    func: Callable[Concatenate[MatterEntity, P], Coroutine[Any, Any, _R]],
+) -> Callable[Concatenate[MatterEntity, P], Coroutine[Any, Any, _R]]:
+    """Catch Matter errors and convert to Home Assistant error."""
+
+    @functools.wraps(func)
+    async def wrapper(self: MatterEntity, *args: P.args, **kwargs: P.kwargs) -> _R:
+        """Catch Matter errors and convert to Home Assistant error."""
+        try:
+            return await func(self, *args, **kwargs)
+        except MatterError as err:
+            error_msg = str(err) or err.__class__.__name__
+            raise HomeAssistantError(error_msg) from err
```

Comment:
Can we be more specific here and maybe raise specific errors?

Let's also add tests for this

Reply:
What more specific do you want ? We are already handling the Matter errors so all we need is turn them into Home Assistant errors and copy the details. If a MatterError is raised, it means its raised by our matter server and not an unexpected, unhandled exception.

Reply:
tests added